plate_mapper/tests/test_plate_mapper.py

```diff
@@ -48,6 +49,66 @@ def test_plate_mapper(self):
             exp = f.read()
         self.assertEqual(obs, exp)
 
+        # test error when column headers are not incremental integers
+        input_f = open(join(datadir, 'plate_map_cherr.txt'), 'r')
+        with self.assertRaises(ValueError) as context:
+            plate_mapper(input_f, None, None)
+        err = 'Error: column headers are not incremental integers.'
+        self.assertEqual(str(context.exception), err)
+
+        # test error when row headers are not in alphabetical order
+        input_f = open(join(datadir, 'plate_map_rherr.txt'), 'r')
+        with self.assertRaises(ValueError) as context:
+            plate_mapper(input_f, None, None)
+        err = 'Error: row headers are not letters in alphabetical order.'
+        self.assertEqual(str(context.exception), err)
+
+        # test a successful conversion with special sample definitions
+        # a plate map file containing normal and special samples
+        input_f = open(join(datadir, 'plate_map_w_special.txt'), 'r')
+        barseq_f = open(join(datadir, 'barseq_temp.txt'), 'r')
+        obs_output_fp = join(self.wkdir, 'obs_mapping.txt')
+        output_f = open(obs_output_fp, 'w')
+        # special sample definition file
+        special_f = open(join(datadir, 'special_samples.txt'), 'r')
+        # expected output mapping file with special samples correctly treated
+        exp_output_fp = join(datadir, 'exp_mapping_w_special.txt')
+        plate_mapper(input_f, barseq_f, output_f, special_f=special_f)
+        with open(obs_output_fp, 'r') as f:
+            obs = f.read()
+        with open(exp_output_fp, 'r') as f:
+            exp = f.read()
+        self.assertEqual(obs, exp)
+
+        # test error when special sample definitions are invalid
+        input_fp = join(datadir, 'plate_map_w_special.txt')
+        barseq_fp = join(datadir, 'barseq_temp.txt')
+        special_fp = join(self.wkdir, 'special_samples_w_error.txt')
+        # no property following sample
+        with open(special_fp, 'w') as f:
+            f.write('#\n+\tPOS\tpositive control\n')
+        with self.assertRaises(ValueError) as context:
```

Comment:
That sounds a good function! Thanks for letting me know. However documents told me that `assertRaisesRegex` is only for Python 3.2+, but this script is meant to be executed on any machine with unspecified Python (usually Python 2.x).